### PR TITLE
test: Add tests for comment association

### PR DESCRIPTION
**Pull request by Zuplyx**
- For #2482

I've created failing UnitTests for some of the issues with comment association reported in the linked issue.

The tests demonstrate the following issues:

- Comments at `CtArrayAccess` and `CtBinaryOperator` elements are not added to the AST.
- Depending on their position, comments at `CtTypeParameter` elements are either wrongly added to the class (comments 1 and 4) or not added to the AST at all (comments 2 and 3).
- Comments at  `CtAnnotation` elements are not added to the annotation, but instead to the `CtLiteral` parameter of the annotation.

### CODE REVIEW

File: src/test/java/spoon/test/comment/CommentTest.java [hunk 1]
```diff
@@ -24,44 +24,8 @@
...
-import spoon.reflect.declaration.CtEnumValue;
-import spoon.reflect.declaration.CtField;
-import spoon.reflect.declaration.CtInterface;
-import spoon.reflect.declaration.CtMethod;
-import spoon.reflect.declaration.CtNamedElement;
-import spoon.reflect.declaration.CtPackage;
-import spoon.reflect.declaration.CtParameter;
-import spoon.reflect.declaration.CtType;
-import spoon.reflect.declaration.ModifierKind;
+import spoon.reflect.code.*;
```
Comment (I-Al-Istannen) [inline src/test/java/spoon/test/comment/CommentTest.java:27]: Could you minify the diff please and keep explicitly importing these?

**Review by I-Al-Istannen — CHANGES_REQUESTED** [review]
Hey, thanks for the tests :) Just a small nitpick.

**Comment by I-Al-Istannen** [comment]
Also paging @SirYwell because I think he has thrown himself off that cliff before :)

File: src/test/java/spoon/test/comment/CommentTest.java (hunk 1 above)
Comment (Zuplyx) [reply src/test/java/spoon/test/comment/CommentTest.java]: I've changed it back to explicit imports.

Comment (I-Al-Istannen) [reply src/test/java/spoon/test/comment/CommentTest.java]: Thank you :)

**Review by I-Al-Istannen — APPROVED** [review]

**Review by SirYwell — APPROVED** [review]
If I ever looked at comments before, I successfully forgot about it again. But the tests look good.

**Comment by I-Al-Istannen** [comment]
Thanks @Zuplyx :)

**Pull request merged**